tests/unit/test_plugins_loader.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import mock
+
+import pkg_resources
+
+import pytest  # NOQA
+
+from gordon import plugins_loader
+
+
+#####
+# Fixtures
+#####
+class FakePlugin:
+    def __init__(self, config):
+        self.config = config
+
+
+@pytest.fixture(scope='session')
+def plugins():
+    plugins = {}
+    names = ['one.plugin', 'two.plugin']
+    for name in names:
+        plugin_mock = mock.MagicMock(pkg_resources.EntryPoint, autospec=True)
+        plugin_mock.name = name
+        plugin_mock.load.return_value = FakePlugin
+        plugins[name] = plugin_mock
+    return plugins
+
+
+@pytest.fixture(scope='session')
+def namespaced_config():
+    return {
+        'one': {'a_key': 'a_value', 'b_key': 'b_value'},
+        'one.plugin': {'a_key': 'another_value'},
+        'two': {},
+        'two.plugin': {'d_key': 'd_value'}
+    }
+
+
+@pytest.fixture(scope='session')
+def plugin_config():
+    return {
+        'one.plugin': {
+            'a_key': 'another_value',
+            'b_key': 'b_value'
+        },
+        'two.plugin': {
+            'd_key': 'd_value'
+        },
+    }
+
+
+@pytest.fixture(scope='session')
+def exp_inited_plugins(plugin_config):
+    return [
+        FakePlugin(plugin_config.get('one.plugin')),
+        FakePlugin(plugin_config.get('two.plugin'))
+    ]
+
+
+@pytest.fixture
+def mock_iter_entry_points(monkeypatch, plugins):
+    mock_plugins = plugins.values()
+
+    mock_iter_entry_points = mock.MagicMock(pkg_resources.iter_entry_points)
+    mock_iter_entry_points.return_value = iter(mock_plugins)
+    monkeypatch.setattr(
+        plugins_loader.pkg_resources, 'iter_entry_points',
+        mock_iter_entry_points)
+
+
+#####
+# Args & Parameters for parametrized tests
+#####
+merge_args = 'namespace,exp_config'
+merge_params = [
+    ('one', {'a_key': 'a_value', 'b_key': 'b_value'}),
+    ('one.plugin', {'a_key': 'another_value', 'b_key': 'b_value'}),
+    ('two', {}),
+    ('two.plugin', {'d_key': 'd_value'}),
+]
+namespace_args = 'namespace,exp_config'
+namespace_params = [
+    ('one', {'a_key': 'a_value', 'b_key': 'b_value'}),
+    ('one.plugin', {'a_key': 'another_value'}),
+    ('two', {}),
+    ('two.plugin', {'d_key': 'd_value'}),
+]
```

Comment:
I would prefer keeping these close to the tests that use them. Separating these out from the tests makes it harder to see what each test is actually testing.